src/comgr.cpp

```diff
@@ -273,7 +273,7 @@ static void RemoveCommonOptionsUnwanted(OptionList& list)
                                 || (option == "-hc")
                                 || (option == "-x hip") || (option == "-xhip")
                                 || (option == "--hip-link")
-                                || (option == "-lclang_rt.builtins-x86_64")
+                                || (option.find("builtins-x86_64") != std::string::npos)
```

Comment:
This check might still cause some issues in future.
clang_rt.builtins lib's name itself is under change in LLVM.
currently its clang_rt.builtins-x86_64, they are planning to move it under a folder called x86_64 and calling the lib just 'clang_rt.builtins'
We should check for 'clang_rt.builtins' instead of 'builtins-x86_64'

Reply:
@cjatin Thanks for reviewing. This sournds reasonable. But is it possible that in the future, some `clang_rt.builtins` (that resides in some other folder) will contain _device_ code?

Reply:
I am not sure, I think @yxsamliu can answer it.

Reply:
@cjatin 
> This check might still cause some issues in future.

I'm afraid the immediate cause of the issues in the future will be changes in the compiler :smile:

But seriously speaking, it would be nice to fetch something like `HIP_COMPILER_FLAGS_DEVICE_ONLY` from hip::device.

Reply:
Resolved.